hack/generate-antrea-samples.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import yaml
+
+if len(sys.argv) == 1:
+    print(" Usage: %s <input yaml file>..." % sys.argv[0], file=sys.stderr)
+    exit(1)
+
+version = os.getenv('ANTREA_VERSION')
+if version is None or version == "" or version == "main":
+    version = "latest"
+
+out = {
+    'apiVersion': 'operator.antrea.vmware.com/v1',
+    'kind': 'AntreaInstall',
+    'metadata': {
+        'name': 'antrea-install',
+        'namespace': 'antrea-operator'},
+    'spec': {}}
+
+for f in sys.argv[1:]:
+    with open(f, "r") as  s:
+        docs = yaml.load_all(s, Loader=yaml.Loader)
+
+        for doc in docs:
+            if doc and doc.get('kind') == 'ConfigMap' and doc.get('metadata', {}).get('name') == 'antrea-config':
+                out['spec']['antreaAgentConfig'] = doc.get('data', {}).get('antrea-agent.conf', "")
+                out['spec']['antreaCNIConfig'] = doc.get('data', {}).get('antrea-cni.conflist', "")
+                out['spec']['antreaControllerConfig'] = doc.get('data', {}).get('antrea-controller.conf', "")
+                out['spec']['antreaImage'] = 'antrea/antrea-ubi:v' + version
```

Comment:
The CRD also has a field `spec.antreaPlatform` with value being `kubernetes` or `openshift`, not sure whether we should add the field when generating sample cr.

[https://github.com/vmware/antrea-operator-for-kubernetes/blob/main/config/crd/operator.antrea.vmware.com_antreainstalls.yaml#L58](https://github.com/vmware/antrea-operator-for-kubernetes/blob/main/config/crd/operator.antrea.vmware.com_antreainstalls.yaml#L58)

Reply:
I see. So the files which are committed under config/* and bundle/*, which are products, can be either for kuberenetes or openshift?
So maybe we can should store them both elsewhere (maybe under deploy/) as we could have two separate products, depending on the platform?

Reply:
For files under config/*, I think the field could be kubernetes or openshift. 
As for those under under bundle/*, I think the filed could only be openshift because only openshift requires `bundle`.
And we actually store them both, in `deploy/kubernetes/cr.yaml`, platform is `kubernetes`. While in `deploy/openshift/cr.yaml`, platform is `openshift`.
[https://github.com/vmware/antrea-operator-for-kubernetes/blob/main/deploy/kubernetes/operator.antrea.vmware.com_v1_antreainstall_cr.yaml#L200](https://github.com/vmware/antrea-operator-for-kubernetes/blob/main/deploy/kubernetes/operator.antrea.vmware.com_v1_antreainstall_cr.yaml#L200)

Maybe we could add simply a parameter for this script like `out['spec']['platform'] = sys.argv[x]` so that we can specify platform when generating cr to different locations?

Reply:
Makes sense, I was just concerned about the location of the outputs. As the make action always stores them under bundle, right? So under config we will have some random result of the last execution, either kubernetes or openshift?

Reply:
Yes, for those under config, I'm ok with random result.